File: src/test/java/org/dedda/bratwurst/lang/StringConcatenationTest.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.ScopedTestCase;
import org.dedda.bratwurst.lang.scope.Scope;
import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
public class StringConcatenationTest extends ScopedTestCase {

    @Test
    public void testRun() throws Exception {
        String[] texts = {
                "Hello ",
                "World",
                "!"
        };
        int intVal = 123;
        Scope scope = createEmptyScope();
        scope.setVariable(new BWVariable("var1", new BWString("Hello ")));
        scope.setVariable(new BWVariable("var2", new BWString("World")));
        scope.setVariable(new BWVariable("var3", new BWString("!")));
        scope.setVariable(new BWVariable("var4", new BWInteger(123)));
        StringConcatenation concatenation = new StringConcatenation(0, new String[]{
                "var1",
                "var2",
                "var3",
                "var4"
        });
        String expected = "Hello World!123";
        concatenation.run(scope);
        String actual = ((BWString) concatenation.getValue()).getStringValue();
        assertEquals(expected, actual);
    }
}

File: src/test/java/org/dedda/bratwurst/lang/StringConcatenationTest.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.ScopedTestCase
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
class StringConcatenationTest : ScopedTestCase() {
    @Test
    @Throws(Exception::class)
    fun testRun() {
        val scope = createEmptyScope()
        scope.setVariable(BWVariable("var1", BWString("Hello ")))
        scope.setVariable(BWVariable("var2", BWString("World")))
        scope.setVariable(BWVariable("var3", BWString("!")))
        scope.setVariable(BWVariable("var4", BWInteger(123)))
        val concatenation = StringConcatenation(0, arrayOf(
                "var1",
                "var2",
                "var3",
                "var4"
        ))
        val expected = "Hello World!123"
        concatenation.run(scope)
        val actual = (concatenation.value as BWString).stringValue
        Assert.assertEquals(expected, actual)
    }
}